### PR TITLE
[BUG] KNN distance params 

### PR DESCRIPTION
**Pull request by chrisholder**
#### Reference Issues/PRs
Issue #446 

#### What does this implement/fix? Explain your changes.
There was a bug where custom distance params were not being passed through to the distance function in the knn and knnregressor. These have been fixed.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.
- [x] For new estimators, I've added the estimator to the online documentation.

### CODE REVIEW

**Review by TonyBagnall — APPROVED** [review]
I'm going to take a look at MSM here, but thanks for fixing it so quickly.

File: setup.cfg [hunk 1]
```diff
@@ -11,15 +11,15 @@ addopts =
...
-    -n auto
+;    --doctest-modules
+;    --durations 10
+;    --timeout 600
+;    --cov aeon
+;    --cov-report xml
+;    --cov-report html
+;    --showlocals
+;    --matrixdesign True
+;    -n auto
```
Comment (MatthewMiddlehurst) [inline setup.cfg:22]: this needs to be reverted

File: aeon/classification/distance_based/tests/test_time_series_neighbors.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "euclidean": 19,
     "dtw": 21,
     "wdtw": 21,
-    "msm": 20,
+    "msm": 10,
```
[the remark this answers is not in the document]
Comment (MatthewMiddlehurst) [reply aeon/classification/distance_based/tests/test_time_series_neighbors.py:36]: ah nvm, this is what it was previously right?

[the remark this answers is not in the document]
Comment (MatthewMiddlehurst) [reply aeon/classification/distance_based/tests/test_time_series_neighbors.py:36]: We should have caught this when it changed then, really 🙂

[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/classification/distance_based/tests/test_time_series_neighbors.py:36]: chris thought a bug he had fixed had improved msm, turns out its just terrible when tuned on this problem!

File: setup.cfg (hunk 1 above)
Comment (TonyBagnall) [reply setup.cfg:22]: thats my trick!

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**